service/controller/chart/v1/resource/release/resource.go

```diff
@@ -78,8 +78,8 @@ func (r *Resource) Name() string {
 	return Name
 }
 
-// Equals asseses the equality of ReleaseStates with regards to distinguishing fields.
-func (a *ReleaseState) Equals(b ReleaseState) bool {
+// equals asseses the equality of ReleaseStates with regards to distinguishing fields.
+func (a *ReleaseState) equals(b ReleaseState) bool {
```

Comment:
I'm not a fan of adding methods to data structs in general.

When it was a function it shouldn't be exported. When this is now a method I believe it should exported. I think private methods of struct should be used only by the struct instance itself (there are exceptions when the struct is a part of the public API).

This is my PoV. Not a blocker.

Reply:
Ack. I converted the methods into functions. I think this is nicer and it aligns the Release and Status resources.